services/dedup/scylla/scylla.go

```diff
@@ -0,0 +1,93 @@
+package scylla
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/gocql/gocql"
+	"github.com/samber/lo"
+
+	"github.com/rudderlabs/rudder-go-kit/config"
+	"github.com/rudderlabs/rudder-go-kit/stats"
+	"github.com/rudderlabs/rudder-server/services/dedup/types"
+)
+
+type ScyllaDB struct {
+	scylla         *gocql.Session
+	conf           *config.Config
+	stat           stats.Stats
+	ttl            int
+	batchSize      int
+	createTableMap map[string]struct{}
+}
+
+func (d *ScyllaDB) Close() {
+	d.scylla.Close()
+}
+
+func (d *ScyllaDB) Set(kv types.KeyValue) (bool, int64, error) {
```

Comment:
This function name doesn't justify what it is doing, its not setting anything, its just checking from the DB. Also Don't we need a in memory cache like we have in badger?

Reply:
https://github.com/rudderlabs/rudder-server/pull/4922#discussion_r1694819117

Didn't want to change the signature from the original contract but can change it to Get since it makes more sense

Reply:
We will need cache... Will make the changes